diff --git a/setup.py b/setup.py
index 8e3b4c3..d19f3c9 100644
--- a/setup.py
+++ b/setup.py
@@ -1,14 +1,17 @@
 #!/usr/bin/env python
-
+import sys
 
 def get_config_schema():
     from aksetup_helper import (ConfigSchema,
             BoostLibraries, Switch, StringListOption, make_boost_base_options)
-
+    if sys.version_info.major >= 3:
+      boost_python = "python3"
+    else:
+      boost_python = "python"
     return ConfigSchema(make_boost_base_options() + [
-        BoostLibraries("python"),
+        BoostLibraries(boost_python),
 
-        Switch("USE_SHIPPED_BOOST", True, "Use included Boost library"),
+        Switch("USE_SHIPPED_BOOST", False, "Use included Boost library"),
 
         StringListOption("CXXFLAGS", [],
             help="Any extra C++ compiler options to include"),
@@ -26,6 +29,7 @@ def main():
 
     hack_distutils(what_opt=1)
     conf = get_config(get_config_schema())
+    print(conf)
 
     TRI_EXTRA_OBJECTS, TRI_EXTRA_DEFINES = \
             set_up_shipped_boost_if_requested("meshpy", conf)
@@ -44,7 +48,10 @@ def main():
 
     INCLUDE_DIRS = conf["BOOST_INC_DIR"] + ["src/cpp"]
     LIBRARY_DIRS = conf["BOOST_LIB_DIR"]
-    LIBRARIES = conf["BOOST_PYTHON_LIBNAME"]
+    if sys.version_info.major >= 3:
+      LIBRARIES = conf["BOOST_PYTHON3_LIBNAME"]
+    else:
+      LIBRARIES = conf["BOOST_PYTHON_LIBNAME"]
 
     init_filename = "meshpy/__init__.py"
     exec(compile(open(init_filename, "r").read(), init_filename, "exec"), conf)
