diff --git a/aksetup_helper.py b/aksetup_helper.py
index a63576f..3d0c499 100644
--- a/aksetup_helper.py
+++ b/aksetup_helper.py
@@ -504,10 +504,7 @@ class Libraries(StringListOption):
 class BoostLibraries(Libraries):
     def __init__(self, lib_base_name, default_lib_name=None):
         if default_lib_name is None:
-            if lib_base_name == "python":
-                default_lib_name = "boost_python-py%d%d" % sys.version_info[:2]
-            else:
-                default_lib_name = "boost_%s" % lib_base_name
+            default_lib_name = "boost_%s" % lib_base_name
 
         Libraries.__init__(self, "BOOST_%s" % lib_base_name.upper(),
                 [default_lib_name],
